tycho-plugins/repository-utils/src/main/java/org/jboss/tools/tycho/sitegenerator/CompareWithBaselineMojo.java

```diff
@@ -0,0 +1,128 @@
+import org.eclipse.tycho.core.osgitools.DefaultReactorProject;
+import org.eclipse.tycho.core.shared.TargetEnvironment;
+import org.eclipse.tycho.osgi.adapters.MavenLoggerAdapter;
+import org.eclipse.tycho.p2.resolver.facade.P2ResolutionResult;
+import org.eclipse.tycho.p2.resolver.facade.P2ResolutionResult.Entry;
+import org.eclipse.tycho.p2.resolver.facade.P2Resolver;
+import org.eclipse.tycho.p2.resolver.facade.P2ResolverFactory;
+import org.eclipse.tycho.p2.target.facade.TargetPlatformConfigurationStub;
+import org.osgi.framework.Version;
+
+@Mojo(defaultPhase=LifecyclePhase.VERIFY, requiresProject=false, name="compare-version-with-baselines")
+public class CompareWithBaselineMojo extends AbstractMojo {
+
+	@Parameter(property="project", readonly=true)
+	private MavenProject project;
+
+	@Parameter(property="baselines", name="baselines")
+	private List<String> baselines;
+
+	@Parameter(property="skip")
+	private boolean skip;
+
+	@Requirement
+	@Component
+	private EquinoxServiceFactory equinox;
+
+	@Component
+	private Logger plexusLogger;
+
+	public void execute() throws MojoExecutionException, MojoFailureException {
+		if (skip) {
+			getLog().info("Skipped");
+			return;
+		}
+		ReactorProject reactorProject = DefaultReactorProject.adapt(project);
+		Set<?> dependencyMetadata = reactorProject.getDependencyMetadata(true);
+		if (dependencyMetadata == null || dependencyMetadata.isEmpty()) {
+			getLog().info("No p2 artifacts created");
+			return;
+		}
+
+		P2ResolverFactory resolverFactory = this.equinox.getService(P2ResolverFactory.class);
+		P2Resolver resolver = resolverFactory.createResolver(new MavenLoggerAdapter(this.plexusLogger, true));
+
+		TargetPlatformConfigurationStub baselineTPStub = new TargetPlatformConfigurationStub();
+		baselineTPStub.setEnvironments(Collections.singletonList(TargetEnvironment.getRunningEnvironment()));
+		for (String baselineRepo : this.baselines) {
+			baselineTPStub.addP2Repository(toRepoURI(baselineRepo));
+		}
+		TargetPlatform baselineTP = resolverFactory.getTargetPlatformFactory().createTargetPlatform(baselineTPStub, new ExecutionEnvironmentConfigurationStub("JavaSE-1.8"), null, null);
+
+		for (Object item : dependencyMetadata) {
+			try {
+				Method getIdMethod = item.getClass().getMethod("getId");
+				Method getVersionMethod = item.getClass().getMethod("getVersion");
+				String id = (String)getIdMethod.invoke(item);
+				Version version = new Version(getVersionMethod.invoke(item).toString());
+				P2ResolutionResult res = resolver.resolveInstallableUnit(baselineTP, id, "0.0.0");
+
+				for (Entry foundInBaseline : res.getArtifacts()) {
+					Version baselineVersion = new Version(foundInBaseline.getVersion());
+					if (version.compareTo(baselineVersion) <= 0) {
+						throw new MojoFailureException("Version of '" + id + '/' + version
+								+ ") must be bigger than baseline one (" + baselineVersion.toString() + ")");
```

Comment:
This reads a bit weird in the console.

Suggestion for more explicit messages as at https://github.com/maxandersen/jbosstools-maven-plugins/commit/70bb67bd3f297215e1e834557eb269c13d5841c1